### PR TITLE
Stats: Adding a gradient to Subscribers chart

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to #75932

## Proposed Changes

* adding a gradient to Subscribers chart

## Testing Instructions

* Open the live branch for this PR
* Navigate to the traffic stats page at `/stats/day/:siteSlug`
* Add the query string `?flags=stats/subscribers-section` to enable the subscriber counts page
* Verify that the chart has a gradient under the main line
* Check if it looks fine when switching to `Month` or `Year` periods

| Before | After |
|---|---|
| SCR-20230426-mkvb | SCR-20230426-mkqv |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by kangzj — APPROVED** [review]
It works well for me. It'd be even better if we add some tests and more comments, which definitely is not a blocker.

File: packages/components/src/chart-uplot/index.tsx [hunk 1]
```diff
@@ -49,16 +49,126 @@ interface UplotChartProps {
...
 }: UplotChartProps ) {
 	const translate = useTranslate();
 	const uplot = useRef< uPlot | null >( null );
 	const uplotContainer = useRef( null );
+	const defaultFillColor = 'rgba(48, 87, 220, 0.075)';
+
+	const can = document.createElement( 'canvas' );
+	const ctx = can.getContext( '2d' );
+
+	const scaleGradient = (
```
Comment (jsnmoon) [inline packages/components/src/chart-uplot/index.tsx:69]: Can we refactor this into a hook? Like `useScaleGradient`?

If not, could we either memorize this function declaration or move the function declaration outside of the component function?

**Review by jsnmoon — APPROVED** [review]
I'm here to eat my words from [this comment](https://github.com/Automattic/wp-calypso/issues/75932#issuecomment-1522605812)... This PR works as described and looks gorgeous while doing so.

image

I left a non-blocking comments inline regarding the maintainability of the chart component.

File: packages/components/src/chart-uplot/index.tsx [hunk 2]
```diff
@@ -49,16 +49,126 @@ interface UplotChartProps {
...
 export default function UplotChart( {
 	data,
 	options: propOptions,
 	legendContainer,
+	solidFill = false,
 }: UplotChartProps ) {
 	const translate = useTranslate();
 	const uplot = useRef< uPlot | null >( null );
 	const uplotContainer = useRef( null );
+	const defaultFillColor = 'rgba(48, 87, 220, 0.075)';
```
Comment (jsnmoon) [inline packages/components/src/chart-uplot/index.tsx:64]: Just thinking: What can we do here to prepare this component for multiple data series in the future? An array of default fill colors, perhaps?

**Comment by grzegorz-cp** [comment]
Thanks for the reviews. I created new issues to improve this feature.

**Pull request merged**